### PR TITLE
CASMCMS-9081: Add the authorization token back into the request.

### PR DESCRIPTION
**Pull request by jsollom-hpe**
## Summary and Scope

Add the authorization token back into the request. It was mistakenly removed in a previous modification.

## Issues and Related PRs

* Resolves [CASMCMS-9081](https://jira-pro.it.hpe.com:8443/browse/CASMCMS-9081)

### Tested on:

  * `Gamora`

### Test description:

I built the rpm containing the bos-reporter. I installed it into on a running node and saw that it successfully report the node's state to the BOS API. 

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why? No, not as a part of my testing.
- Was upgrade tested? If not, why? No. This was a fix not an upgrade.
- Was downgrade tested? If not, why? No.
- Were new tests (or test issues/Jiras) created for this change? No.

## Risks and Mitigations

None. It's already broken.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by mharding-hpe — APPROVED** [review]
Minor linting nits

File: src/bos/reporter/client.py [hunk 1]
```diff
@@ -67,10 +68,19 @@ def get_auth_token(path='/opt/cray/auth-utils/bin/get-auth-token'):
             LOGGER.error(
                 'get_auth_token failed to retrieve authorization token: code=%d: error=%s',
                 e.returncode, e.output)
-        except Exception:
-            LOGGER.exception('Unexpected exception')
+        except Exception as e:
+            LOGGER.exception(f"Unexpected exception: {e}")
```
Comment (mharding-hpe) [inline src/bos/reporter/client.py:72]: pylint also dislikes the use of f-strings in calls to logging methods, preferring lazy evaluation, but I think that mainly matters for logging calls that may not end up in the log due to the logging level that is set -- and I suspect this one would always make it through that filter, so...

**Comment by mharding-hpe** [comment]
I believe this will need backports to `release/2.10` and `support/2.0`

File: src/bos/reporter/client.py (hunk 1 above)
Comment (jsollom-hpe) [reply src/bos/reporter/client.py:71]: Not sure if I should shift this back or not.

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**